ros2nodl/ros2nodl/_verb/_show.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2020 Canonical Ltd.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License Version 3 as published by
+# the Free Software Foundation.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import pprint
+import shutil
+import sys
+
+import nodl
+from ros2cli.verb import VerbExtension
+from ros2pkg.api import package_name_completer, PackageNotFoundError
+from ros2run.api import ExecutableNameCompleter
+
+
+class _ShowVerb(VerbExtension):
+    """Show NoDL data."""
+
+    def add_arguments(self, parser: argparse.ArgumentParser, cli_name: None = None):
+        # Ignoring type because of https://github.com/python/typeshed/issues/1878
+        parser.add_argument(  # type: ignore
+            'package_name', help='Name of the package to show.'
+        ).completer = package_name_completer
+
+        command_group = parser.add_mutually_exclusive_group(required=True)
+        command_group.add_argument(
+            '-a', '--all', action='store_true', help='Show all executables in package.',
+        )
+        # Ignoring type because of https://github.com/python/typeshed/issues/1878
+        command_group.add_argument(  # type: ignore
+            'file', nargs='*', default=[], help='Specific Executable to display.'
+        ).completer = ExecutableNameCompleter(package_name_key='package_name')
+
+    def main(self, args: argparse.Namespace) -> int:
+        package = args.package_name
+        try:
+            nodes = nodl._index._get_nodes_from_package(package_name=package)
```

Comment:
It seems inefficient to do this before we know we need this below, no?

Reply:
Fixed in 1d45ece